src/engine/strat_engine/cmd.rs

```diff
@@ -232,13 +232,77 @@ fn get_clevis_executable(name: &str) -> StratisResult<PathBuf> {
     }
 }
 
+// Return the mkfs.xfs version reported by the "-V" option as a string.
+fn get_mkfs_xfs_version() -> StratisResult<String> {
+    let mut command = Command::new(get_executable(MKFS_XFS).as_os_str())
+        .arg("-V")
+        .stdout(Stdio::piped())
+        .spawn()?;
+    command.wait()?;
+
+    let mut output = String::new();
+    let is_ok = command.wait()?.code() == Some(0);
+    command
+        .stdout
+        .ok_or_else(|| {
+            StratisError::Msg("Could not read string value from xfs.mkfs output".to_string())
+        })?
+        .read_to_string(&mut output)?;
+
+    if is_ok {
+        output
+            .trim()
+            .strip_prefix("mkfs.xfs version ")
+            .ok_or_else(|| {
+                StratisError::Msg("Could not parse version string from mkfs.xfs output".to_string())
+            })
+            .map(|v| v.to_string())
+    } else {
+        Err(StratisError::Msg(
+            "\"mkfs.xfs -V\" returned an error".to_string(),
+        ))
+    }
+}
+
 /// Create a filesystem on devnode. If uuid specified, set the UUID of the
-/// filesystem on creation. If 'noalign', set the noalign option.
-/// NOTE: It used to be desirable to set the noalign option when creating the
-/// MDV, because it was very small; since then the size of the MDV has been
-/// increased, and it is no longer necessary to pass the noalign option when
-/// creating the MDV.
-pub fn create_fs(devnode: &Path, uuid: Option<StratisUuid>, noalign: bool) -> StratisResult<()> {
+/// filesystem on creation.
+pub fn create_fs(devnode: &Path, uuid: Option<StratisUuid>) -> StratisResult<()> {
+    // If the version can not be obtained, which is unlikely, assume that the
+    // version of mkfs.xfs is new enough to use the nrext64 option. This will
+    // become more and more true with time. If the version is not high enough,
+    // this will be obvious when the command-line invocation fails.
+    // The check only verifies that the major number is greater than 5, because
+    // before that time, the larger extent counter size was not used by default,
+    // so, for earlier versions, it is unnecessary to turn it off explicitly.
+    // Choose not to use a semantic version package, because there is no Rust
+    // package availabe that necessarily adheres to the xfsprogs versioning
+    // scheme.
+    let use_nrext64_option = match get_mkfs_xfs_version().and_then(|v| {
+        v.split('.')
```

Comment:
We may not want to add this now, but this crate would simplify this code a lot. https://docs.rs/semver/latest/semver/